src/main/java/org/gridsuite/merge/orchestrator/server/NetworkConversionService.java

```diff
@@ -45,21 +68,121 @@ public NetworkConversionService(RestTemplate restTemplate) {
         this.networkConversionServerRest = restTemplate;
     }
 
-    public ExportNetworkInfos exportMerge(List<UUID> networksIds, String format, String baseFileName) {
-        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION + "/networks/{networkUuid}/export/{format}");
+    public void setCaseFetcherService(CaseFetcherService caseFetcherService) {
+        this.caseFetcherService = caseFetcherService;
+    }
+
+    public void setCgmesBoundaryService(CgmesBoundaryService cgmesBoundaryService) {
+        this.cgmesBoundaryService = cgmesBoundaryService;
+    }
+
+    public FileInfos exportMerge(List<UUID> networkUuids, List<UUID> caseUuids, String format, String baseFileName) throws IOException {
+        if (format.equals(CGMES_FORMAT)) {
+
+            List<FileInfos> cgmesProfiles = new ArrayList<>();
+
+            //Add merged IGMs profiles
+            List<FileInfos> mergedIgms = caseFetcherService.getCases(caseUuids);
+            for (FileInfos mergedIgm : mergedIgms) {
+                cgmesProfiles.addAll(unzipCgmes(mergedIgm));
+            }
+
+            //Add SV profile
+            cgmesProfiles.add(getSvProfile(networkUuids, baseFileName));
+
+            //Add boundary files
+            cgmesProfiles.addAll(getBoundaries());
+
+            //Zip files
+            ByteArrayOutputStream baosZip = createZipFile(cgmesProfiles);
+
+            return new FileInfos(baseFileName.concat(UNDERSCORE + FILE_VERSION + ZIP), baosZip.toByteArray());
+        } else {
+            UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION + "/networks/{networkUuid}/export/{format}");
```

Comment:
don't use UriComponentsBuilder, instead create a template that has "networkUuid={n1}&networkUuid={n2}..." and pass the networkUuids as a variable to RestTemplate.exchange()

Reply:
Ah this is from the previous code .. You can leave it then, we will fix it later